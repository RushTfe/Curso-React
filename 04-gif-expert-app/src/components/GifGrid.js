import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from '../components/GifGridItem'
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

	const { data:images, loading } = useFetchGifs( category );

  	return (
		<>
			<h3 className='animate__animated animate__fadeIn'>{ category }</h3>

			{ loading && <p className='animate__animated animate__flash'>Loading</p> }

			<div className='card-grid'>
				{
					images.map( img => <GifGridItem key={ img.id }{ ...img } />)
				}
			</div>
		</>

  )
}

const getImagesInfo = (images) => images.map(({id, title}) => <li key={id}>{title}</li>);

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
}