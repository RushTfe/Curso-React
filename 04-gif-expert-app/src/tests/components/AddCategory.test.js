import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en Add Category', () => { 
    

    const setCategories = jest.fn();
    let wrapper;


    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    });


    test('Debe mostrarse correctamente', () => { 
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', {target: {value: value}});

        const parrafo = wrapper.find('p');

        expect( parrafo.text().trim() ).toBe( value );


    });

    test('No debe de postear la informaciÃ³n en el submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Debe de llamar al setCategories y limpiar las cajas de texto', () => {
        const value = 'Hola Mundo';
        const obj = { target: { value: value } };

        // 1. Simular input change
        wrapper.find('input').simulate('change', obj);

        // 2. Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

        expect( wrapper.find('input').prop('value')).toBe('');


    })

 });