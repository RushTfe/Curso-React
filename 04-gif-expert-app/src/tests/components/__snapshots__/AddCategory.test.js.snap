// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Add Category Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Escribe un tema para buscar"
    type="text"
    value=""
  />
</form>
`;
